test/types/middleware.preposttypes.test.ts

```diff
@@ -693,51 +652,6 @@ schema.post('deleteOne', { document: false, query: false }, function(res) {
   expectNotType<Query<any, any>>(res);
 });
 
-schema.pre(['save', 'init'], function() {
-  expectType<HydratedDocument<IDocument>>(this);
-});
-
-schema.post(['save', 'init'], function(res) {
-  expectType<HydratedDocument<IDocument>>(this);
-  expectNotType<Query<any, any>>(res);
-});
-
-schema.pre(['save', 'init'], { document: true, query: false }, function() {
-  expectType<HydratedDocument<IDocument>>(this);
-});
-
-schema.post(['save', 'init'], { document: true, query: false }, function(res) {
-  expectType<HydratedDocument<IDocument>>(this);
-  expectNotType<Query<any, any>>(res);
-});
-
-schema.pre(['save', 'init'], { document: true, query: true }, function() {
-  expectType<HydratedDocument<IDocument>>(this);
-});
-
-schema.post(['save', 'init'], { document: true, query: true }, function(res) {
-  expectType<HydratedDocument<IDocument>>(this);
-  expectNotType<Query<any, any>>(res);
-});
-
-schema.pre(['save', 'init'], { document: false, query: true }, function() {
-  expectType<never>(this);
-});
-
-schema.post(['save', 'init'], { document: false, query: true }, function(res) {
-  expectType<never>(this);
-  expectNotType<Query<any, any>>(res);
-});
-
-schema.pre(['save', 'init'], { document: false, query: false }, function() {
-  expectType<never>(this);
-});
-
-schema.post(['save', 'init'], { document: false, query: false }, function(res) {
-  expectType<never>(this);
-  expectNotType<Query<any, any>>(res);
-});
```

Comment:
I didnt write about this before, but should those maybe also be restored for just `save`-array case? (which is possible, though weird and should likely never be used)